Server/serverSlate.go

```diff
@@ -67,18 +67,22 @@ func runOsCommand(cmd *exec.Cmd) int {
 	return 1
 }
 
-func simulateKeyPress(key string, keyPressCount int, timeStamps []string) {
-	timeStamps[0] = strconv.Itoa(int(time.Now().UnixNano() / 1000000))
+func simulateKeyPress(key string, keyPressCount int, timeStamps []string, keyPressSequence []string) {
 	time.Sleep(1 * time.Second)
-	for i := 1; i <= keyPressCount; i++ {
+	runningSeq := ""
+	for i := 0; i < keyPressCount; i++ {
+		androidTime, _ := exec.Command("adb", "shell", "date", "+%s%3N").Output()
```

Comment:
Previously at this point, I was querrying current server UTC time and storing them. They were later sent to android device and used in lag calculation. As the server clock was not in sync with android, a mechanism to find the drift between two clocks were also implemented.(and used in lag calculation)

In this new implementation, we querry the current UTC time of android device and store them. They are later sent to android device and used in lag calculation. As there is only a single clock now, there is no need to find drift now.

Benchmarking the command, found execution time of command as
```
real    0m0.108s
user    0m0.006s
sys     0m0.014s
```
This can be used as an offset to further sync received androidTime. _(Some discussion is required on this end)_

Reply:
Thanks Rohit. I think this is an important improvement. 

I have some doubts as to how reliable this synchronisation method will be (longer-term), but let's catch up offline about this -- I know you are already aware of better synchronisation options.

Reply:
100ms execution time means this error from this synchronization is on the same order as what we are trying to measure right?

Reply:
I realised that the command used here is not the most efficient one.
When I enter the android shell using `adb shell` and inside it I can see:
```shell
RMX1971:/ $ time date +%s%3N                                                                  
1597121530314
    0m00.04s real     0m00.02s user     0m00.02s system
```

In previous benchmark the time of opening android shell got added.

Reply:
As discussed about this approach, due the significant overheads of using adb connection this way, this approach is dropped.